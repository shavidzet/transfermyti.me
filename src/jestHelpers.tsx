/* eslint-disable import/no-extraneous-dependencies,@typescript-eslint/no-explicit-any,no-console */
import React from "react";
import {
  RenderOptions,
  RenderResult,
  render as testRender,
} from "@testing-library/react";
import { ThemeProvider } from "emotion-theming";
import { theme } from "./components/emotion";

// These are exported for convenience (so we don't have to add an additional import
// to the test file).
export { createEvent, fireEvent, RenderResult } from "@testing-library/react";
export { theme } from "./components/emotion";

/**
 * Renders the specified component with react-testing-library wrapped in a theme
 * provider so we don't have to do it explicitly for each component.
 */
export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
): RenderResult {
  return testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options);
}

type LogMethod = "log" | "warn" | "error";

export class ConsoleToggle {
  private consoleMethod: (...args: any[]) => void = console.warn;
  private readonly logMethod: LogMethod;

  public constructor(logMethod: LogMethod) {
    this.logMethod = logMethod;
  }

  public onBefore(): void {
    this.consoleMethod = console[this.logMethod];
    console[this.logMethod] = jest.fn();
  }

  public onAfter(): void {
    console[this.logMethod] = this.consoleMethod;
  }
}
