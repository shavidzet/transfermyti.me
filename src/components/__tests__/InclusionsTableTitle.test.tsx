import React from "react";
import { fireEvent, render, RenderResult } from "~/jestHelpers";
import InclusionsTableTitle from "../InclusionsTableTitle";

const setup = (
  propOverrides: any = {},
): { props: any; wrapper: RenderResult } => {
  const props = {
    children: "Test Title",
    flipDisabled: false,
    onFlipAreAllIncluded: jest.fn(),
    ...propOverrides,
  };

  const wrapper = render(<InclusionsTableTitle {...props} />);

  return { props, wrapper };
};

describe("the <InclusionsTableTitle> component", () => {
  test("matches its snapshot with valid props", () => {
    const { wrapper } = setup();

    expect(wrapper.container.firstElementChild).toMatchSnapshot();
  });

  test("does not disable toggle inclusions button if props.flipDisabled is undefined", () => {
    const { wrapper } = setup({ flipDisabled: undefined });

    expect(wrapper.getByRole("button")).not.toHaveAttribute("disabled", "");
  });

  test("disables toggle inclusions button if props.flipDisabled = true", () => {
    const { wrapper } = setup({ flipDisabled: true });

    expect(wrapper.getByRole("button")).toHaveAttribute("disabled", "");
  });

  test("fires props.onFlipAreAllIncluded when toggle inclusions is clicked", () => {
    const { wrapper, props } = setup({ flipDisabled: false });
    fireEvent.click(wrapper.getByRole("button"));

    expect(props.onFlipAreAllIncluded).toHaveBeenCalled();
  });
});
