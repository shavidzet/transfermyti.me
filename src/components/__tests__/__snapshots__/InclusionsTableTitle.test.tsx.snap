// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <InclusionsTableTitle> component matches its snapshot with valid props 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-1 button {
  border-radius: 0.25rem;
  font-size: 0.875rem;
  padding: 0.25rem 0.5rem;
}

.emotion-1 button {
  font-weight: 600;
}

.emotion-0 {
  border-radius: 0.375rem;
  font-size: 1.25rem;
  font-weight: 400;
  padding: 0.5rem 0.75rem;
  text-align: center;
  -webkit-transition: 0.3s all linear;
  transition: 0.3s all linear;
  vertical-align: middle;
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.2), 0 1px 1px 0 rgba(0,0,0,0.14), 0 2px 1px -1px rgba(0,0,0,0.12);
  background: #EFFDE8;
  border: 3px solid #1E78A1;
  color: #1E78A1;
  opacity: 1;
}

.emotion-0:hover {
  background: #EFFDE8;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus {
  outline-color: rgb(30,120,161);
}

<div
  class="emotion-1"
>
  <h4>
    Test Title
  </h4>
  <button
    aria-disabled="false"
    class="emotion-0"
    type="button"
  >
    Include All/None
  </button>
</div>
`;
